llvm_passes/HipPrintf.cpp

```diff
@@ -18,158 +53,363 @@
 #include "llvm/Passes/PassBuilder.h"
 #include "llvm/Passes/PassPlugin.h"
 
+#include <vector>
+#include <string>
+#include <iostream>
+
+#define SPIRV_OPENCL_CONSTANT_AS 2
+#define SPIRV_OPENCL_GENERIC_AS 4
```

Comment:
These constants could be pulled from SPIR-V headers (https://github.com/KhronosGroup/SPIRV-Headers).